pathway-analysis/ssgsea_example.Rmd

```diff
@@ -0,0 +1,392 @@
+  # x-axis text at a 45 degree angle to increase readability
+  ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 45, hjust = 1))
+```
+
+```{r}
+# saving to file
+ggplot2::ggsave(file.path("plots", "GSE75574_ssgsea_comp_coag_facet.pdf"))
+```
+
+Looks like there may be some differences between tissues in this pathway, with
+some strains showing more pronounced differences in white adipose tissue in the 
+calorie-restricted vs. control comparison.
+We should always follow up with more formal analysis (e.g., QuSAGE), but 
+this series of steps is a good way to explore one's data.
+
+#### Aside: Gene set _size_ influences ssGSEA score values
+
+ssGSEA score values are not necessarily comparable between gene sets of 
+different sizes (number of genes).
+If you are comparing ssGSEA scores between samples _within a gene set_, this is
+not a concern.
+
+To demonstrate the effect of gene set size on scores, we'll perform a short 
+experiment with random gene sets.
+
+```{r}
+# use all the gene symbols in the dataset as the pool of possible genes
+all_genes <- symbol_exprs_df$SYMBOL
+
+# set a seed for reproducibility
+set.seed(123)
+
+# GSVA::gsva takes gene sets as a list, so that's how we'll store the random
+# gene sets
+random_gene_sets <- list()
+
+# testing 5 different gene set sizes
+for (pathway_size in c(25, 50, 100, 250, 500)) {
+  # generate 10 random sets of pathway_size
+  for (path_iter in 1:10) {
+    random_pathway_name <- paste("size", pathway_size, path_iter, sep = "_")
+    current_gene_set <- base::sample(x = all_genes, size = pathway_size)
+    random_gene_sets[[random_pathway_name]] <- current_gene_set
+  }
+}
+
+# calculate ssGSEA scores for the random gene sets
+random_ssgsea_results <- gsva(expr = exprs_mat,
+                              gset.idx.list = random_gene_sets,
+                              method = "ssgsea",
+                              verbose = FALSE)
+
+# plot the results with ggplot2
+# first we need to get this in a form that is amenable for plotting, as gsva
+# returns a matrix where columns are samples and rows are gene sets
+random_long_df <- as.data.frame(random_ssgsea_results) %>%
+  # gene set names are rownames
+  tibble::rownames_to_column("gene_set") %>%
+  # long format
+  reshape2::melt(variable.name = "sample", value.name = "ssgsea_score") %>%
+  # extract the gene set size from the gene set name
+  dplyr::mutate(gene_set_size = stringr::word(gene_set, 2, sep = "_")) %>%
+  # we want to plot smallest no. genes -> largest no. genes
+  dplyr::mutate(gene_set_size = factor(gene_set_size, 
+                                       levels = c(25, 50, 100, 250, 500)))
+
+# violin plot comparing no. genes / gene set size
+random_long_df %>%
+  ggplot2::ggplot(ggplot2::aes(y = ssgsea_score, x = gene_set_size)) +
+  ggplot2::geom_violin() +
+  ggplot2::coord_flip() +
+  ggplot2::labs(title = "Random gene set scores",
+                x = "gene set size",
+                y = "ssGSEA score")
```

Comment:
Did you wanna save this plot to the `plots` folder too? Also adding you could try adding a `theme_classic` to get rid of the ugly gray grid.

Reply:
I'll add `theme_bw()` but I'm not going to save this for the same reason as above -- want this to be an aside.